### PR TITLE
Add patch releases support to Jenkins pipeline

### PR DESCRIPTION
**Pull request by israel-hdez**
Fixes kiali/kiali#3687

### CODE REVIEW

**Comment by israel-hdez** [comment]
* This is a run for a `minor` release with these changes: https://github.com/israel-hdez/helm-charts/commit/2b1d26cbbc10847dfee946b6db583eefb06ba731
* This is a run for a patch release: https://github.com/israel-hdez/helm-charts/commit/59d611ccb910876209fde11c4df255020ae8658f

File: Jenkinsfile [hunk 1]
```diff
@@ -77,20 +82,47 @@ node('kiali-build && fedora') {
...
+        //
+        // This change to the Makefile is not done for patch releases, because that
+        // would break the next minor build (remember we have already switched to master).
+        sh "sed -i -r 's/^VERSION \\?= v.*/VERSION \\?= v${releasingVersion}/' Makefile"
+      }
+
+      sh "make -e VERSION=v${releasingVersion} update-helm-repos"
 
       // Tag the release
+      //   Note that if we are doing a patch release, this tag won't contain valid `kiali-server` nor `kiali-operator` directories
```
Comment (israel-hdez) [inline Jenkinsfile:116]: I think we should no longer create tags. They don't seem to be that useful. For patch releases, the tag will only contain the updated index.yaml and the tarballs, which will already be in master.

Probably, only the vX.Y branch is enough. But I'm not sure...

Comment (jmazzitelli) [reply Jenkinsfile:116]: I like tags for historical purposes. We can see exactly which commit produced which patch build. So at least we have it for historical and documentation purposes.

Comment (jmazzitelli) [reply Jenkinsfile:116]: Note - I did not follow the code exactly, but when I say I like the tag for historical purposes, the tag I mean is the "vX.Y.Z" tag that is created on the git commit where the build-helm-repos is run.

**Review by jmazzitelli — APPROVED** [review]
looks ok

**Comment by israel-hdez** [comment]
@jmazzitelli I also need your approval on this other PR: https://github.com/kiali/kiali/pull/3710

**Comment by jmazzitelli** [comment]
> @jmazzitelli I also need your approval on this other PR: [kiali/kiali#3710](https://github.com/kiali/kiali/pull/3710)

done

**Pull request merged**